devenv/scripts/push_addon_image.sh

```diff
@@ -0,0 +1,25 @@
+# Get cluster information for az aks command invoke
```

Comment:
Do you think we could split this PR into one thats changing the current E2E testing and then a follow up on enabling private clusters?

Reply:
not exactly. combining both regular and private zone e2e testing relies on making the public suite use specific functionality (namely, the `az aks command invoke` command for deploying/monitoring everything). I could in theory try merging one PR for public stuff and then just rewrite whatever needs to be rewritten to make it compatible with private, but that doesn't seem like the smartest way to move forward imo.

Reply:
update: after discussion, switching to this approach of merging public e2e changes to minimize size of this PR